ibis_vega_transform/vegaexpr.py

```diff
@@ -112,126 +162,161 @@ def now() -> float:
     return ibis.now()
 
 
-def datetime(
-    year: int,
-    month: int,
-    day: int = 0,
-    hour: int = 0,
-    min: int = 0,
-    sec: int = 0,
-    millisec: int = 0,
-) -> dt.datetime:
-    """Returns a new Date instance.
-    The month is 0-based, such that 1 represents February.
+def z(v) -> bool:
     """
-    # TODO: do we need a local timezone?
-    return dt.datetime(
-        int(year),
-        int(month) + 1,
-        int(day),
-        int(hour),
-        int(min),
-        int(sec),
-        int(millisec * 1000),
-    )
+    Returns whether the value is equal to a literal 0
+    """
+    return not isinstance(v, ibis.expr.types.Expr) and v == 0
 
 
-def date(datetime: dt.datetime) -> int:
+def datetime(y, m, d, h, mi, s, ms):
+    """
+    This should use https://github.com/ibis-project/ibis/issues/386 if it gets implemented
+
+    Currently we only support use cases that can be turned into a truncate, which is:
```

Comment:
How serious do you think the `truncate` limitation is? Some example charts I've been playing with need non-truncation versions of this function.

Reply:
I'm wondering if we can support arbitrary timestamps here by using ibis timestamp literals, and then adding them together.

Reply:
>  Some example charts I've been playing with need non-truncation versions of this function.

Could you point to an example, so I could take a look at it? I think we could get creative depending on the particular use case to try to work around other cases of it.

> I'm wondering if we can support arbitrary timestamps here by using ibis timestamp literals, and then adding them together.

Not sure I follow this, what would the resulting SQL look like? If omnisci had a `datetime` operator, then we wouldn't need any hacks here. cc @alexbaden who I think was looking into this